tv_apps/dashboard/js/digital_clock.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+(function (exports) {
+
+  function DigitalClock () {
+    this.date = null;
+
+    this.timeoutId = null;
+    this.intervalId = null;
+
+    this.elements = {
+      'hours-tens-digit': document.getElementById('hours-tens-digit'),
+      'hours-units-digit': document.getElementById('hours-units-digit'),
+      'minutes-tens-digit': document.getElementById('minutes-tens-digit'),
+      'minutes-units-digit': document.getElementById('minutes-units-digit'),
+      'seconds-tens-digit': document.getElementById('seconds-tens-digit'),
+      'seconds-units-digit': document.getElementById('seconds-units-digit'),
+      'weekday': document.getElementById('weekday'),
+      'month': document.getElementById('month'),
+      'date': document.getElementById('date')
+    };
+  }
+
+  DigitalClock.prototype = {
+
+    init: function () {
+      this._initTime(new Date());
+    },
+
+    start: function () {
+      this._start(this._update.bind(this));
+    },
+
+    stop: function () {
+      if (this.timeoutId) {
+        window.clearTimeout(this.timeoutId);
+        this.timeoutId = null;
+      }
+
+      if (this.intervalId) {
+        window.clearInterval(this.intervalId);
+        this.intervalId = null;
+      }
+    },
+
+    /**
+     * Start ticking the clock.
+     * @param  {Function} refresh A callback to run on every tick.
+     */
+    _start: function (refresh) {
+      var now = new Date();
+
+      this.timeoutId = setTimeout(function () {
+        refresh(new Date());
+
+        this.intervalId = setInterval(function () {
+          refresh(new Date());
+        }, 1000);
+      }.bind(this), 1000 - now.getMilliseconds());
+    },
+
+    /**
+     * Init the time of the clock.
+     */
+    _initTime: function (date) {
+      var hours = date.getHours();
+      var minutes = date.getMinutes();
+      var seconds = date.getSeconds();
+
+      this.elements['hours-tens-digit'].textContent = Math.floor(hours / 10);
```

Comment:
Isn't it plenty to just call  `_update()`? These are just duplicated codes with that.